src/components/token/TokenBalances.js

```diff
@@ -5,372 +5,367 @@
+   *         Sort is an array with two strings, The value is given by ElasticSearch and it is passed back when we want to change page
+   * hasAfter: Indicates if a next page exists
+   * hasBefore: Indicates if a previous page exists
+   * sortBy: Which field to sort (uid, name, symbol)
+   * order: If sorted field must be ordered asc or desc
+   * page: Current page. Used to know if there is a previous page
+   * pageSearchAfter: Calculates the searchAfter param that needs to be passed to explorer-service in order to get the next/previous page.
+   *                  We use this array to store already-calculated values,
+   *                  so we do not need to recalculate them if user is requesting an already-navigated page.
+   * loading: Initial loading, when user clicks on the Tokens navigation item
+   * isSearchLoading: Indicates if search results are being retrieved from explorer-service
+   * calculatingPage: Indicates if next page is being retrieved from explorer-service
+   * error: Indicates if an unexpected error happened when calling the explorer-service
+   * tokenBalanceInformationError: Indicates if an unexpected error happened when calling the token balance information service
+   * maintenanceMode: Indicates if explorer-service or its downstream services are experiencing problems. If so, maintenance mode will be enabled on
+   *                  our feature toggle service (Unleash) to remove additional load until the team fixes the problem
+   * transactionsCount: Number of transactions for the searched token
+   * addressesCount: Number of addressed for the searched token
+   * tokensApiError: Flag indicating if the request to the token api failed, to decide wether to display or not the total number of transactions
    */
-  pagination = new PaginationURL({
-    sortBy: { required: false },
-    order: { required: false },
-    token: { required: false },
-  });
-
-  constructor(props) {
-    super(props);
+  const [tokenId, setTokenId] = useState(hathorLibConstants.NATIVE_TOKEN_UID);
+  const [tokenBalances, setTokenBalances] = useState([]);
+  const [hasAfter, setHasAfter] = useState(false);
+  const [hasBefore, setHasBefore] = useState(false);
+  const [sortBy, setSortBy] = useState(null);
+  const [order, setOrder] = useState(null);
+  const [page, setPage] = useState(1);
+  const [pageSearchAfter, setPageSearchAfter] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [isSearchLoading, setIsSearchLoading] = useState(false);
+  const [calculatingPage, setCalculatingPage] = useState(false);
+  const [error, setError] = useState(false);
+  const [tokenBalanceInformationError, setTokenBalanceInformationError] = useState(false);
+  const [transactionsCount, setTransactionsCount] = useState(0);
+  const [addressesCount, setAddressesCount] = useState(0);
+  const [tokensApiError, setTokensApiError] = useState(false);
```

Comment:
[note] I don't know if this is a good idea because it can cause the component to trigger an unnecessary amount of re-renderings and it can cause other problems.

Reply:
React batches state changes made within a single synchronous iteration, so there will be only a single re-render for all those changes ( except on some very specific async situations, not the case in this component ).

I found [this article](https://medium.com/swlh/react-state-batch-update-b1b61bd28cd2) that explains it way better than I could, with details, example codes and live tests.